directory.cpp

```diff
@@ -9,154 +9,126 @@ directory::directory(string name, directory *parent):ancestor(name,parent)
 
 directory::~directory()
 {
-    for(auto i:subdirectories){
+    for(auto i:FileSystemObjects){
         delete i;
     }
 }
 
-list<ancestor *> directory::getSubdirectories() const
+list<ancestor *> directory::getFileSystemObjects() const
 {
-    return subdirectories;
+    return FileSystemObjects;
 }
 
 directory *directory::getparent()
 {
-    return this->parent;
+    directory* temp=dynamic_cast<directory*>(this->parent);
+    return temp;
 }
 
-bool directory::isDir() const
-{
-    return true;
-}
-
-
 string directory::getname()
 {
     return this->name;
 }
 
 void directory::ls(){
-    for(ancestor* i:subdirectories){
+    for(ancestor* i:FileSystemObjects){
         cout<<i->getname()<<endl;
     }
 }
 
 bool directory::hasDirs()
 {
-    if(this->getSubdirectories().empty()){
+    if(this->getFileSystemObjects().empty()){
         return false;
     }
     return true;
 }
 
 void directory::makefolder(string n)
 {
-    subdirectories.push_back(new directory(n,this));
+    FileSystemObjects.push_back(new directory(n,this));
 }
 
 void directory::touch(string name)
 {
-    subdirectories.push_back(new file(name,this));
+    FileSystemObjects.push_back(new file(name,this));
 }
 
 void directory::rm(string todelete)
 {
-    for(auto it=subdirectories.begin();it!=subdirectories.end();){
-        directory* temp=dynamic_cast<directory*>(*it);
-        if(temp->getname()==todelete &&!temp->hasDirs()){
-            it=this->subdirectories.erase(it);
-        }
-        else if(temp->getname()==todelete &&temp->hasDirs()){
-            cout<<"Az adott mappa nem ures"<<endl;
-            it++;
+    std::list<ancestor*>::iterator i = FileSystemObjects.begin();
+    while (i != FileSystemObjects.end())
+    {
+        if((*i)->getname()==todelete){
+            directory* temp=dynamic_cast<directory*>(*i);
+            if(temp!=nullptr){
+            //ha üres a mappa kitöröljük és továbbiterálunk
+                if(!temp->hasDirs())
+                    FileSystemObjects.erase(i++);
+                //ha nem üres meghívjuk a segedrmrf-et ami rekurzívan bejárja az adott mappát
+                else{
+                    cout<<"Az adott mappa nem ures"<<endl;
+                    ++i;
```

Comment:
de itt minek mesz meg tovabb? MEg fentebb is ha toroltel, meg lentebb is ha fajlt toroltel?